apps/server/doc/testing.md

```diff
@@ -208,8 +215,10 @@ describe('somefunction', () => {
 		});
 	});
 });
-
 ```
+For creating specific mock implementations the helper functions which only mock the implementation once, should be used (e.g. mockReturnValueOnce). With that approach more control over mocked functions can be achieved.
+
+At times there could be the need to mock functions only for a specific test. In that case we strongly recommend the use of `jest.spyOn` and not `jest.fn`, because `jest.spyOn` can be restored a lot easier. 
```

Comment:
You mean functions that are not part of a mocked dependency, don't you? You could state that explicitly.

Reply:
"At times there could be the need to mock a function only for a specific test without mocking the entire dependency." Like that?

Reply:
I ask myself what functions are those... Basically (or only?) those of the module under test, or not? Are there other cases?

Your wording sounds like: "If you just need a mock for one test, then you don't need to mock the dependency. You can just mock the function in place." But that's not the case because you always mock all dependencies anyway, don't you?